src/celeritas/em/model/WentzelModel.cc

```diff
@@ -0,0 +1,1881 @@
+{
+    Applicability electron_applic;
+    electron_applic.particle = this->host_ref().ids.electron;
+    electron_applic.lower = detail::coulomb_scattering_limit();
+    electron_applic.upper = detail::high_energy_limit();
+
+    Applicability positron_applic = electron_applic;
+    positron_applic.particle = this->host_ref().ids.positron;
+
+    return {electron_applic, positron_applic};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get the microscopic cross sections for the given particle and material.
+ */
+auto WentzelModel::micro_xs(Applicability applic) const -> MicroXsBuilders
+{
+    return imported_.micro_xs(std::move(applic));
+}
+
+//---------------------------------------------------------------------------//
+//!@{
+/*!
+ * Apply the interaction kernel.
+ */
+void WentzelModel::execute(CoreParams const& params, CoreStateHost& state) const
+{
+    auto execute = make_action_track_executor(
+        params.ptr<MemSpace::native>(),
+        state.ptr(),
+        this->action_id(),
+        InteractionApplier{WentzelExecutor{this->host_ref()}});
+    return launch_action(*this, params, state, execute);
+}
+
+//---------------------------------------------------------------------------//
+#if !CELER_USE_DEVICE
+void WentzelModel::execute(CoreParams const&, CoreStateDevice&) const
+{
+    CELER_NOT_CONFIGURED("CUDA OR HIP");
+}
+#endif
+//!@}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get the model ID for this model.
+ */
+ActionId WentzelModel::action_id() const
+{
+    return this->host_ref().ids.action;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Load Mott coefficients and construct per-element data.
+ */
+void WentzelModel::build_data(HostVal<WentzelData>& host_data,
+                              MaterialParams const& materials)
+{
+    // Build element data
+    unsigned int const num_elements = materials.num_elements();
+    auto elem_data = make_builder(&host_data.elem_data);
+    elem_data.reserve(num_elements);
+
+    for (auto el_id : range(ElementId{num_elements}))
+    {
+        // Load Mott coefficients
+        WentzelElementData z_data;
+        z_data.mott_coeff
+            = get_mott_coeff_matrix(materials.get(el_id).atomic_number());
+
+        elem_data.push_back(z_data);
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Interpolated Mott coefficients used by the Lijian, Quing, Zhengming
+ * expression [PRM 8.48] for atomic numbers 1 <= Z <= 92.
+ * This data was taken from Geant4's G4MottData.hh file.
+ *
+ * For higher Z values, the PRM cites a numerical solution by Boschini
+ * et al (2013), but does not implement it.
+ */
+WentzelElementData::MottCoeffMatrix
+WentzelModel::get_mott_coeff_matrix(AtomicNumber z)
+{
+    using CoeffMat = WentzelElementData::MottCoeffMatrix;
+    CELER_EXPECT(z);
```

Comment:
Preconditions are easier to read if they're the first part of the function
```suggestion
    CELER_EXPECT(z);

    using CoeffMat = WentzelElementData::MottCoeffMatrix;
```